raiden_contracts/utils/merkle.py

```diff
@@ -0,0 +1,57 @@
+from collections import namedtuple
+from typing import Iterable
+from itertools import zip_longest
+from eth_utils import keccak
+
+
+EMPTY_MERKLE_ROOT = b'\x00' * 32
+
+MerkleTree = namedtuple('MerkleTree', ['layers'])
+
+
+def _hash_pair(first: bytes, second: bytes) -> bytes:
+    """ Computes the hash of the items in lexicographic order """
+    if first is None:
+        return second
+
+    if second is None:
+        return first
+
+    if first > second:
+        return keccak(second + first)
+    else:
+        return keccak(first + second)
+
+
+def compute_merkle_tree(items: Iterable[bytes]) -> MerkleTree:
+    """ Calculates the merkle root for a given list of items """
+
+    if not all(isinstance(l, bytes) and len(l) == 32 for l in items):
+        raise ValueError('Not all items are hashes')
+
+    leaves = sorted(items)
+    if len(leaves) == 0:
+        return MerkleTree(layers=[[EMPTY_MERKLE_ROOT]])
+
+    if not len(leaves) == len(set(leaves)):
+        raise ValueError('The items must not cointain duplicate items')
+
+    tree = [leaves]
+    layer = leaves
+    while len(layer) > 1:
```

Comment:
I recognize this pattern. This is the one we were discussing in the chat in the morning and we have it in the [raiden code](https://github.com/raiden-network/raiden/blob/master/raiden/transfer/merkle_tree.py#L57). Can you make a note somewhere to get it abstracted it and put it in the common raiden libs and use it both in raiden and here.

Reply:
Also opened https://github.com/raiden-network/raiden-contracts/issues/42 to track common components with `raiden-libs` (which I will probably end up using)